nats.go

```diff
@@ -3478,16 +3479,19 @@ func (nc *Conn) Barrier(f func()) error {
 	return nil
 }
 
-// GetCID returns the client ID assigned by the server the client
-// is currently connected to. Note that the value would change if
-// client reconnects. Also, returned value may be 0 if connecting
-// to a server pre 1.2.0, which did not send the CID back to the
-// client.
-func (nc *Conn) GetCID() (uint64, error) {
+// GetClientID returns the client ID assigned by the server to which
+// the client is currently connected to. Note that the value may change if
+// the client reconnects.
+// This function returns ErrNoClientIDReturned if the server is of a
+// version prior to 1.2.0.
+func (nc *Conn) GetClientID() (uint64, error) {
 	nc.mu.Lock()
 	defer nc.mu.Unlock()
 	if nc.isClosed() {
 		return 0, ErrConnectionClosed
 	}
+	if nc.info.CID == 0 {
```

Comment:
Then the server should make sure that if it wraps, then it start again to 1.
I guess we could search "client_id" in the info string we get before doing unmarshal and if not present have a flag in connection that says cid not set?

Reply:
Maybe post an issue against the server.

Reply:
Ok, is it really worth it? I mean at 10M new connections a second, it would take 58,494 years without restarting the server before it wraps (if my math is correct).

Reply:
Agree.

Reply:
Ok, so will change the error name/content and update